### PR TITLE
KEYCLOAK-18547 Add tolerations and nodeSelector in Keycloak deployment

### LINKED ISSUE #408: Allow Node Selector and Tolerations to be customised
### Description

We run tainted nodes which require us to annotate our pods, having this facility in the operator would be very helpful

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by david6983**
Fixes #408

## Additional Information

- As discussed in the [mailing list](https://groups.google.com/g/keycloak-dev/c/JYAV7BeAFHI), I didn't implemented the features on the PostgreSQL deployment
- This issue is related with the pull request https://github.com/keycloak/keycloak-operator/pull/401
- I decided to create a new Pull Request to start from a cleaner branch since the previous was forked on the master branch and I struggled making a rebase
- I moved the **tolerations** and the **nodeSelector** in `KeycloakDeploymentSpec` as requested in the code review
- I made similar changed in [rhsso_deployment](https://github.com/keycloak/keycloak-operator/blob/c24c55ac3be47a379c553c1ff9c87a8e175d7e27/pkg/model/rhsso_deployment.go). Could you please tell me if it is what you requested @vmuzikar ?

## Verification Steps

- A version of the operator has been deployed in production in a Kubernetes cluster with taints on the nodes, the deployment ran succesfully.

## Checklist:
- [X] Automated Tests
- [X] [Documentation](https://github.com/keycloak/keycloak-documentation) changes if necessary
- [X] (If required) Discussed on [Keycloak DEV](https://groups.google.com/forum/#!forum/keycloak-dev)

### CODE REVIEW

File: pkg/apis/keycloak/v1alpha1/keycloak_types.go
Comment (vmuzikar) [inline pkg/apis/keycloak/v1alpha1/keycloak_types.go:109]: Thinking about this more, this should actually go into `Experimental` where we already `Affinity` that is related to it.
https://github.com/keycloak/keycloak-operator/blob/1b9a02eb2417e9124ebbf4e9f37ec415d25c6b34/pkg/apis/keycloak/v1alpha1/keycloak_types.go#L133

File: pkg/model/keycloak_deployment.go
Comment (vmuzikar) [inline pkg/model/keycloak_deployment.go:260]: Could you please add similar change to:
https://github.com/keycloak/keycloak-operator/blob/1b9a02eb2417e9124ebbf4e9f37ec415d25c6b34/pkg/model/keycloak_deployment.go#L281

File: pkg/model/rhsso_deployment.go
Comment (vmuzikar) [inline pkg/model/rhsso_deployment.go:210]: Could you please add similar change to:
https://github.com/keycloak/keycloak-operator/blob/1b9a02eb2417e9124ebbf4e9f37ec415d25c6b34/pkg/model/rhsso_deployment.go#L232

File: pkg/model/postgresql_deployment.go
Comment (vmuzikar) [inline pkg/model/postgresql_deployment.go:146]: The embedded DB is [deprecated](https://www.keycloak.org/docs/latest/release_notes/index.html#deprecated-features-in-the-keycloak-operator). We should not do any changes to it. Could you please remove it?

**Review by vmuzikar — CHANGES_REQUESTED** [review]
@david6983 Thank you for your contribution! Added a few last comments. Could you please also add a simple test?

**Comment by vmuzikar** [comment]
@david6983 Just wanted to ask if this is still something you'd like to work on.

**Comment by rdelcampog** [comment]
Hope requested changes can be performed soon as we really need tolerations for our use-case :(

**Comment by david6983** [comment]
@vmuzikar I would be glad to continue this PR but I always struggle to edit the PR because I need to rebase. Last time I did this, it didn't worked well

**Comment by david6983** [comment]
@vmuzikar Let me know what you think of the changes. I tried to rebase, could you tell me if I did it properly?

**Comment by vmuzikar** [comment]
@david6983 Thank you for your response. There's something weird going on with the commits. I would suggest to create a new branch, cherry-pick your changes and force-push to this remote branch.

**Comment by patlachance** [comment]
@david6983  / @vmuzikar what is needed to merge this PR?

**Comment by david6983** [comment]
I just updated the PR, and need approval to launch the tests then if everything ok maintainers can merge

>

**Comment by vmuzikar** [comment]
Thank you very much for you contribution. Keycloak 19 was the last version that included this legacy Operator, and with the release of Keycloak 20 the Operator reached EOL, please see our [blog post](https://www.keycloak.org/2022/03/releases.html) on this topic. Since this PR was not ready before Keycloak 19.0.0 had been released (and we're not adding new features in micro releases), I'm closing this PR as it is no longer relevant for this Operator (or for the [Realm Operator](https://github.com/keycloak/keycloak-realm-operator) which doesn't include Keycloak CR). Thanks for your understanding. And be sure to check out our [new Operator](https://www.keycloak.org/guides#operator)!

**Pull request closed without merging**
